test/unit/gateways/credorax_test.rb

```diff
@@ -276,6 +276,18 @@ def test_supports_billing_descriptor
     end.respond_with(successful_purchase_response)
   end
 
+  def test_adds_processor_fields
```

Comment:
Since this test seems to be specific to the ```purchase``` operation, should the test method be named ```test_purchase_adds_processor_fields``` rather than ```test_adds_processor_fields```?

Reply:
I was just following the convention in this test file, which also includes `test_adds_3d_secure_fields`, `test_adds_3ds2_fields_via_normalized_hash`, `test_adds_default_cavv_when_omitted_from_normalized_hash`, `test_adds_a9_field`, `test_adds_submerchant_id`, and `test_supports_billing_descriptor`, all of which test only the `purchase` transaction type.

Reply:
In this case, I see the lack of test coverage for the non-```purchase``` methods as technical debt rather than a convention that we'd want to follow. Perhaps it was originally an oversight or a conscious decision based on time constraints, but the impact is that we don't have tests to catch regressions that would prevent the GSFs from being added to the request for those operations. Having those kinds of things break isn't uncommon when refactoring code. So, having tests to cover the new code can be quite valuable in allowing us to refactor or make other changes in the future quickly and with confidence.

It's always nice to chip away at technical debt when we can. Virtually any codebase will have some, and Active Merchant is certainly no exception. That said, deciding the best way to spend your time is more of an art. This would be my personal take for this scenario if I were working on the ticket:

* Definitely add the tests to cover the new code that we're adding in this PR, since they are valuable in their own right.

* As an optional bonus, change the test method names that you mentioned to include the word ```purchase```. This makes things better by indicating that they're really only testing the ```purchase``` method, which can be helpful to someone skimming the code, or someone who runs into a test failure. It also addresses part of the inconsistency you noticed, and would be a quick win.

* Skip the task of going back to add test coverage for all of the other GSFs in non-```purchase``` methods. This is out of scope for the original ticket you're working on, and  seems like it would be time-consuming. You likely also wouldn't have the complete context that you have for the new code you actually added here, and there are probably more valuable things that you could do with your time at the moment.

When deciding whether to tackle tech debt that I run into, I usually consider those two factors of context and time. If I can't knock it out relatively quickly given the context that I have from the original work I was tasked with, I tend to leave it for another day. Another thing to keep in mind is that you can always enter a ticket for tech debt tasks that you believe are valuable. That way, we're able to prioritize the time-consuming ones against our other work.

Reply:
The additional unit tests have been added for the gateway specific fields. I haven't added in any additional remote tests since they are still not working and therefore can't be tested.